src/native/libs/System.Security.Cryptography.Native/openssl.c

```diff
@@ -1313,150 +1301,6 @@ int32_t CryptoNative_IsSignatureAlgorithmAvailable(const char* algorithm)
-
-// Set of locks initialized for OpenSSL
-static pthread_mutex_t* g_locks = NULL;
-
-/*
-Function:
-LockingCallback
-
-Called back by OpenSSL to lock or unlock.
-*/
-static void LockingCallback(int mode, int n, const char* file, int line)
-{
-    (void)file, (void)line; // deliberately unused parameters
-
-// Clang complains about releasing locks that are not held.
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wthread-safety-analysis"
-
-#ifndef CRYPTO_LOCK
-#define CRYPTO_LOCK 1
-#endif
-
-    int result;
-    if (mode & CRYPTO_LOCK)
-    {
-        result = pthread_mutex_lock(&g_locks[n]);
-    }
-    else
-    {
-        result = pthread_mutex_unlock(&g_locks[n]);
-    }
-
-    if (result != 0)
-    {
-        assert(0 && "LockingCallback failed.");
-    }
-#pragma clang diagnostic pop
-}
-
-/*
-Function:
-EnsureOpenSslInitialized
-
-Initializes OpenSSL with a locking callback to ensure thread safety.
-
-Return values:
-0 on success
-non-zero on failure
-*/
-static int32_t EnsureOpenSsl10Initialized(void)
-{
-    int ret = 0;
-    int numLocks = 0;
-    int locksInitialized = 0;
-    int randPollResult = 0;
-    size_t allocationSize = 0;
-
-    pthread_mutex_lock(&g_initLock);
-
-    if (g_locks != NULL)
-    {
-        // Already initialized; nothing more to do.
-        goto done;
-    }
-
-    // Determine how many locks are needed
-    numLocks = CRYPTO_num_locks();
-    if (numLocks <= 0)
-    {
-        assert(0 && "CRYPTO_num_locks returned invalid value.");
-        ret = 1;
-        goto done;
-    }
-
-    // Create the locks array
-    if (!multiply_s(sizeof(pthread_mutex_t), (size_t)numLocks, &allocationSize))
-    {
-        ret = 2;
-        goto done;
-    }
-
-    g_locks = (pthread_mutex_t*)malloc(allocationSize);
-    if (g_locks == NULL)
-    {
-        ret = 2;
-        goto done;
-    }
-
-    // Initialize each of the locks
-    for (locksInitialized = 0; locksInitialized < numLocks; locksInitialized++)
-    {
-        if (pthread_mutex_init(&g_locks[locksInitialized], NULL) != 0)
-        {
-            ret = 3;
-            goto done;
-        }
-    }
-
-    // Initialize the callback
-    CRYPTO_set_locking_callback(LockingCallback);
-
-    // Initialize the random number generator seed
-    randPollResult = RAND_poll();
-    if (randPollResult < 1)
-    {
-        ret = 4;
-        goto done;
-    }
-
-    // Load the SHA-2 hash algorithms, and anything else not in the default
-    // support set.
-    OPENSSL_add_all_algorithms_conf();
-
-    // Ensure that the error message table is loaded.
-    ERR_load_crypto_strings();
-
-    // In OpenSSL 1.0.2-, CRYPTO_EX_INDEX_X509 is 10.
-    g_x509_ocsp_index = CRYPTO_get_ex_new_index(10, 0, NULL, NULL, ExDataDupOcspResponse, ExDataFreeOcspResponse);
-    // In OpenSSL 1.0.2-, CRYPTO_EX_INDEX_SSL_SESSION is 3.
-    g_ssl_sess_cert_index = CRYPTO_get_ex_new_index(3, 0, NULL, NULL, ExDataDupNoOp, ExDataFreeNoOp);
```

Comment:
The equivalent of these in the 1.1+ initializer can change from literals to their symbolic name.  Though it looks like I only c_static_asserted CRYPTO_EX_INDEX_X509, we should also do CRYPTO_EX_INDEX_SSL_SESSION